commands/addons.js

```diff
@@ -255,36 +254,45 @@ module.exports = {
 
     run:         cli.command(co.wrap(run)),
     usage:       `${topic} [--all|--app APP]`,
-    description: 'lists your add-ons and attachments',
+    description: 'Lists your add-ons and attachments',
     help:        `The default filter applied depends on whether you are in a Heroku app
 directory. If so, the --app flag is implied. If not, the default of --all
 is implied. Explicitly providing either flag overrides the default
-behaviour.
+behavior.
 
 Examples:
 
   $ heroku ${topic} --all
-  $ heroku ${topic} --app acme-inc-website
-
-Identifying specific add-ons:
-
-  Add-ons have canonical names (ADDON_NAME in these command help texts).
-  They also application-specific names called aliases or attachments. In
-  many cases, these names can be used interchangeably when unambiguous.
-
-  For example, given a fictional \`slowdb\` add-on named \`slowdb-cubed-1531\`
-  attached as \`FOO_DB\` to \`app1\` and BAR_DB to \`app2\`, the following
-  invocations are considered equivalent:
-
-  $ heroku addons:upgrade slowdb             slowdb:premium
-  $ heroku addons:upgrade --app app1 FOO_DB  slowdb:premium
-  $ heroku addons:upgrade app1::FOO_DB       slowdb:premium
-  $ heroku addons:upgrade app2::BAR_DB       slowdb:premium
-  $ heroku addons:upgrade acme-inc-datastore slowdb:premium
-
-  If the name used is ambiguous (e.g. if you used \`slowdb\` with more than
-  one add-on of that type installed), the command will error due to
-  ambiguity and a more specific identifier will need to be chosen.
+  $ heroku ${topic} --app acme-inc-www
+
+Overview of Add-ons:
+
+  Add-ons are created with the \`addons:create\` command, providing a reference
+  to an add-on service (such as \`heroku-postgresql\`) or a service and plan
+  (such as \`heroku-postgresql:hobby-dev\`).
```

Comment:
Added example and reworked the the collective examples so they build on the previous. It's a bit lengthy now, so opportunities to make this more concise are welcome. Some of these may be better suited in the docs for `addons:create`, I think. Maybe we can just explain what names are and only *that* they can be user-specified at creation time ("see help for `addons:create`") or attachment time ("see help for `addons:attach`")...